### PR TITLE
[FLINK-22856][Azure] Upgrade to ubuntu-20.04

### PR DESCRIPTION
**Pull request by rmetzger**
Azure currently shows warnings because the ubuntu-16.04 image will be deprecated in September. This change will update our CI to ubuntu-20.04 (= ubuntu-latest on Azure).
I chose not to use ubuntu-latest to avoid surprises by Azure updating the version underneath.

### CODE REVIEW

**Comment by rmetzger** [comment]
e2e failed on my personal CI, I'll investigate https://dev.azure.com/rmetzger/Flink/_build/results?buildId=9119&view=results&j=9401bf33-03c4-5a24-83fe-e51d75db73ef&t=72901ab2-7cd0-57be-82b1-bca51de20fba

File: tools/azure-pipelines/jobs-template.yml [hunk 1]
```diff
@@ -208,10 +208,10 @@ jobs:
       continueOnError: true
       condition: not(eq(variables['SKIP'], '1'))
     - script: |
-        echo "##vso[task.setvariable variable=JAVA_HOME]$JAVA_HOME_11_X64"
-        echo "##vso[task.setvariable variable=PATH]$JAVA_HOME_11_X64/bin:$PATH"
-      displayName: "Set to jdk11"
-      condition: eq('${{parameters.jdk}}', 'jdk11')
+        echo "##vso[task.setvariable variable=JAVA_HOME]$JAVA_HOME_8_X64"
+        echo "##vso[task.setvariable variable=PATH]$JAVA_HOME_8_X64/bin:$PATH"
+      displayName: "Set to jdk8"
```
Comment (zentol) [inline tools/azure-pipelines/jobs-template.yml:213]: so jdk11 is the default nowadays?

Comment (rmetzger) [reply tools/azure-pipelines/jobs-template.yml:213]: Yes, in Ubuntu 20.04 it seems so

**Review by zentol — APPROVED** [review]

**Comment by rmetzger** [comment]
Thanks for the review.... Merging

**Pull request merged**